From 81b934829881f738df3ec957b9bdd19936a15523 Mon Sep 17 00:00:00 2001
From: czapek1337 <czapek1337@gmail.com>
Date: Thu, 7 Apr 2022 19:39:05 +0200
Subject: [PATCH 2/2] sysdeps/linux: Add sys_getppid sysdep

---
 sysdeps/linux/generic/sysdeps.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sysdeps/linux/generic/sysdeps.cpp b/sysdeps/linux/generic/sysdeps.cpp
index 582115a..bc3f8cd 100644
--- a/sysdeps/linux/generic/sysdeps.cpp
+++ b/sysdeps/linux/generic/sysdeps.cpp
@@ -50,6 +50,7 @@
 #define NR_getgid 104
 #define NR_geteuid 107
 #define NR_getegid 108
+#define NR_getppid 110
 #define NR_rt_sigsuspend 130
 #define NR_sigaltstack 131
 #define NR_arch_prctl 158
@@ -522,4 +523,10 @@ int sys_readlink(const char *path, void *buf, size_t bufsiz, ssize_t *len) {
 	return 0;
 }
 
+pid_t sys_getppid() {
+	auto ret = do_syscall(NR_getppid);
+	// getppid() always succeeds.
+	return sc_int_result<pid_t>(ret);
+}
+
 } // namespace mlibc
-- 
2.35.1

