From 6fa94d65da9680ee96fa050bfc66d9009ef52960 Mon Sep 17 00:00:00 2001
From: czapek1337 <czapek1337@gmail.com>
Date: Sat, 7 May 2022 21:38:40 +0200
Subject: [PATCH] sysdeps/linux: Some fixes to the PR #396

---
 sysdeps/linux/generic/sysdeps.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/sysdeps/linux/generic/sysdeps.cpp b/sysdeps/linux/generic/sysdeps.cpp
index 526f20b..2efab97 100644
--- a/sysdeps/linux/generic/sysdeps.cpp
+++ b/sysdeps/linux/generic/sysdeps.cpp
@@ -435,22 +435,23 @@ int sys_gethostname(char *buf, size_t bufsize) {
 }
 
 int sys_tcgetattr(int fd, struct termios *attr) {
-	auto ret = do_syscall(NR_ioctl, TCGETS, attr);
+	auto ret = do_syscall(NR_ioctl, fd, TCGETS, attr);
 	if (int e = sc_error(ret); e)
 		return e;
 	return 0;
 }
 
-int sys_tcsetattr(int fd, int optional_action, struct termios *attr) {
+int sys_tcsetattr(int fd, int optional_action, const struct termios *attr) {
 	int req;
 
 	switch (optional_action) {
 		case TCSANOW: req = TCSETS; break;
 		case TCSADRAIN: req = TCSETSW; break;
 		case TCSAFLUSH: req = TCSETSF; break;
+		default: return EINVAL;
 	}
 
-	auto ret = do_syscall(NR_ioctl, req, attr);
+	auto ret = do_syscall(NR_ioctl, fd, req, attr);
 	if (int e = sc_error(ret); e)
 		return e;
 	return 0;
-- 
2.35.1

