From 6b2e44c0d1366ca5205f28acea59e71aaf66ca3c Mon Sep 17 00:00:00 2001
From: czapek1337 <czapek1337@gmail.com>
Date: Sun, 8 May 2022 22:59:52 +0200
Subject: [PATCH 2/2] sysdeps/linux: Implement sys_sync and friends

---
 sysdeps/linux/generic/sysdeps.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/sysdeps/linux/generic/sysdeps.cpp b/sysdeps/linux/generic/sysdeps.cpp
index 02974ac..8cd14bb 100644
--- a/sysdeps/linux/generic/sysdeps.cpp
+++ b/sysdeps/linux/generic/sysdeps.cpp
@@ -43,6 +43,8 @@
 #define NR_kill 62
 #define NR_uname 63
 #define NR_fcntl 72
+#define NR_fsync 74
+#define NR_fdatasync 75
 #define NR_getcwd 79
 #define NR_chdir 80
 #define NR_rename 82
@@ -61,6 +63,7 @@
 #define NR_rt_sigsuspend 130
 #define NR_sigaltstack 131
 #define NR_arch_prctl 158
+#define NR_sync 162
 #define NR_sys_futex 202
 #define NR_getdents64 217
 #define NR_clock_gettime 228
@@ -656,4 +659,22 @@ pid_t sys_getppid() {
 	return sc_int_result<pid_t>(ret);
 }
 
+void sys_sync() {
+	do_syscall(NR_sync);
+}
+
+int sys_fsync(int fd) {
+	auto ret = do_syscall(NR_fsync, fd);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_fdatasync(int fd) {
+	auto ret = do_syscall(NR_fdatasync, fd);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
 } // namespace mlibc
-- 
2.35.1

